### PR TITLE
Release 1.4 post

### PR DESCRIPTION
**Pull request by mboersma**
~~Needs a title and summary, probably rewriting of some "What's Next?" sections, and proofreading.~~

~~Needs better summary, rewriting of the app SSL "What's Next?" section, and proofreading.~~

Needs proofreading & suggestions.

### CODE REVIEW

File: _posts/2015-03-03-needs-title.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+## What is Deis?
+
+Deis is an open source PaaS that makes it easy to deploy and manage applications on your own servers. Deis builds upon [Docker](http://docker.io/) and [CoreOS](https://coreos.com/) to provide a lightweight PaaS with a Heroku-inspired workflow.
+
+## 1.4.0 Summary
+
+### New Features
+
+- feat(builder): Try shallow cloning buildpack repos - [#2988](https://github.com/deis/deis/pull/2988)
```
Comment (bacongobbler) [inline _posts/2015-03-03-needs-title.md]: aren't these in more human-readable form in previous release notes? http://deis.io/deis-1-3-code-quality-and-security-enhancements/

File: _posts/2015-03-03-needs-title.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+- feat(builder+docker+user-data): proxy settings - [#2825](https://github.com/deis/deis/pull/2825)
+- feat(contrib): add Microsoft Azure provision scripts and documentation - [#2893](https://github.com/deis/deis/pull/2893)
```
Comment (bacongobbler) [inline _posts/2015-03-03-needs-title.md]: "You can now provision Deis on Microsoft Azure"

Comment (carmstrong) [inline _posts/2015-03-03-needs-title.md]: I think that's going to be the headline this release.

File: _posts/2015-03-03-needs-title.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
+---
+layout: post
+title: "Deis v1.4 - ???"
```
Comment (carmstrong) [inline _posts/2015-03-03-needs-title.md]: Deis v1.4 - Now on Azure

File: _posts/2015-03-03-needs-title.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+Deis is an open source PaaS that makes it easy to deploy and manage applications on your own servers. Deis builds upon [Docker](http://docker.io/) and [CoreOS](https://coreos.com/) to provide a lightweight PaaS with a Heroku-inspired workflow.
+
+## 1.4.0 Summary
+
+### New Features
+
+- feat(builder): Try shallow cloning buildpack repos - [#2988](https://github.com/deis/deis/pull/2988)
+- feat(builder+docker+user-data): proxy settings - [#2825](https://github.com/deis/deis/pull/2825)
+- feat(contrib): add Microsoft Azure provision scripts and documentation - [#2893](https://github.com/deis/deis/pull/2893)
+- feat(contrib/ec2): add support for internal ELBs - [#3074](https://github.com/deis/deis/pull/3074)
```
Comment (bacongobbler) [inline _posts/2015-03-03-needs-title.md]: "The AWS provisioning scrips now have support for internal ELBs"

File: _posts/2015-03-03-needs-title.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+- feat(deisctl): start / stop all installed units - [#3106](https://github.com/deis/deis/pull/3106)
```
Comment (bacongobbler) [inline _posts/2015-03-03-needs-title.md]: "All units which are scalable can be started and stopped with `deisctl start component@*`"

File: _posts/2015-03-03-needs-title.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+- feat(ec2/cloudformation): use SSD EBS volumes by default - [#3065](https://github.com/deis/deis/pull/3065)
```
Comment (bacongobbler) [inline _posts/2015-03-03-needs-title.md]: "AWS now uses Elastic Block Storage volumes by default"

File: _posts/2015-03-03-needs-title.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+- feat(logspout): use custom datetime - [#3037](https://github.com/deis/deis/pull/3037)
```
Comment (bacongobbler) [inline _posts/2015-03-03-needs-title.md]: "Logspout now accepts custom datetime output formats"

File: _posts/2015-03-03-needs-title.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+- feat(router) nginx status; log http_host, upstream and request times - [#2869](https://github.com/deis/deis/pull/2869)
+- feat(router): add optional HTTPs redirect - [#3075](https://github.com/deis/deis/pull/3075)
+- feat(scheduler): graceful shutdown with SIGTERM - [#3091](https://github.com/deis/deis/pull/3091)
```
Comment (bacongobbler) [inline _posts/2015-03-03-needs-title.md]: "Deis gracefully sends SIGTERM when shutting down application processes"

File: _posts/2015-03-03-needs-title.md (hunk 1 above)
Comment (mboersma) [inline _posts/2015-03-03-needs-title.md]: Yes, I just cherry-picked these from what was selected by the contributors update script. I'll resummarize them, thanks for all your suggestions.

Comment (bacongobbler) [inline _posts/2015-03-03-needs-title.md]: np. I started getting carried away so I stopped for now :)

File: _posts/2015-03-03-deis-1-4-now-on-azure.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+- router now logs http_host, upstream and request times - [#2869](https://github.com/deis/deis/pull/2869)
+- router allows optional HTTPS redirect - [#3075](https://github.com/deis/deis/pull/3075)
+- store-gateway can be scaled with `deisctl scale` - [#2929](https://github.com/deis/deis/pull/2929)
+- Any scalable component can be started or stopped with `deisctl start|stop component@` - [#3106](https://github.com/deis/deis/pull/3106)
+- New documentation added about [disk usage](http://docs.deis.io/en/latest/managing_deis/disk_usage/) - [#3021](https://github.com/deis/deis/pull/3021)
+- New links added to Deis [community projects](https://github.com/deis/deis/blob/master/contrib/README.md#community-contributions)! - [#3114](https://github.com/deis/deis/pull/3114)
+
+### Improvements
+
+- builder properly escapes backticks in environment variables - [#3129](https://github.com/deis/deis/pull/3129)
```
Comment (bacongobbler) [inline _posts/2015-03-03-deis-1-4-now-on-azure.md]: we used to "bold-style" the component names. Should we keep doing that for consistency's sake?

File: _posts/2015-03-03-deis-1-4-now-on-azure.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+- builder lets docker-in-docker choose its storage driver - [#3039](https://github.com/deis/deis/pull/3039)
+- controller queries fleet directly for app container state -[#2993](https://github.com/deis/deis/pull/2993)
+- controller removes timed-out fleet units from `deis run` - [#3127](https://github.com/deis/deis/pull/3127)
+- controller kills processes removed from a `Procfile` - [#3068](https://github.com/deis/deis/pull/3068)
+- `deis domains` has improved validation per [RFC 1123](http://www.rfc-editor.org/rfc/rfc1123.txt) - [#2991](https://github.com/deis/deis/pull/2991)
+- `deisctl journal|status` are more helpful about global units - [#3119](https://github.com/deis/deis/pull/3119)
+- router disables SSLv3 (ref: CVE-2014-3566), if SSL is enabled - [#3136](https://github.com/deis/deis/pull/3136)
```
Comment (bacongobbler) [inline _posts/2015-03-03-deis-1-4-now-on-azure.md]: I don't think the comma's necessary here

File: _posts/2015-03-03-deis-1-4-now-on-azure.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+
+### Under the Hood
+
+- [CoreOS](https://coreos.com/releases/) updated to 557.2.0 stable - [#3048](https://github.com/deis/deis/pull/3048)
+- [`go`](http://golang.org/) updated to 1.4.1+ - [#3018](https://github.com/deis/deis/pull/3018)
+- builder updated gradle and play buildpacks - [#2989](https://github.com/deis/deis/pull/2989)
+- controller updated djangorestframework to 3.0.5 - [#3061](https://github.com/deis/deis/pull/3061)
+- controller updated gunicorn to 19.2.1 - [#3024](https://github.com/deis/deis/pull/3024)
+- controller update PostgreSQL driver `psycopg2` to 2.6 - [#3050](https://github.com/deis/deis/pull/3050)
```
Comment (bacongobbler) [inline _posts/2015-03-03-deis-1-4-now-on-azure.md]: updated

File: _posts/2015-03-03-deis-1-4-now-on-azure.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+
+Thanks to community discussion and effort, work on application SSL support has begun. What remains is an agreement on [the formal specification](https://github.com/deis/deis/pull/2911) following Heroku and an updated implementation. Once merged, users will be able to install custom private keys and SSL certificates to expose HTTPS applications using the Deis router.
+
+### Improved Test Infrastructure
+
+We need to speed up test suites, eliminate the PR build queue, and increase test coverage to include operational aspects such as upgrades and HA. Hard work behind the scenes and invaluable community support are steadily improving the Deis test infrastructure.
+
+### New Container Scheduler
+
+Application containers require a resource-aware scheduler to improve cluster efficiency at scale and to alleviate issues with [cluster balancing after the loss of a node](https://github.com/deis/deis/issues/2920). Prototyping is well underway with [Docker Swarm](https://github.com/deis/deis/pull/3134) and Kubernetes.
```
Comment (bacongobbler) [inline _posts/2015-03-03-deis-1-4-now-on-azure.md]: I think "Docker Swarm" sounds a little silly, similar to how "Google Kubernetes" would sound like. We should drop the "Docker" and just call it Swarm.

Comment (bacongobbler) [inline _posts/2015-03-03-deis-1-4-now-on-azure.md]: Also we should link Kubernetes over to deis/deis#2744 so users have somewhere to follow its progress

File: _posts/2015-03-03-deis-1-4-now-on-azure.md [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+- @wenzowski: Rolling restarts with deis ps:restart, SIGTERM on exceeded memory quota, pass env vars to buildpack, feat(store-admin): add gnu watch, builder re-generating SSH keys when jumping hosts, Nginx 5-tuple microflow affinity load balancing to enable arbitrary TCP, Graceful shutdown with SIGTERM, feat(scheduler): graceful shutdown with SIGTERM, fix(router): disable SSLv3 CVE-2014-3566
+- @withinboredom: Failure in deis registry
+- @yekeqiang: command fails: deisctl config platform set sshPrivateKey=~/.ssh/deis
+
+The Deis community continues to grow, and Deis wouldn't be here without you! If we slighted your contribution to this release, please let us know so we can update.
+
+
+## How can you help?
+
+* Star our [GitHub repository](https://github.com/opdemand/deis)
```
Comment (bacongobbler) [inline _posts/2015-03-03-deis-1-4-now-on-azure.md]: link should be fixed to https://github.com/deis/deis

**Comment by bacongobbler** [comment]
Docs LGTM pending comments

File: _posts/2015-03-03-deis-1-4-now-on-azure.md [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+## What is Deis?
+
+Deis is an open source PaaS that makes it easy to deploy and manage applications on your own servers. Deis builds upon [Docker](http://docker.io/) and [CoreOS](https://coreos.com/) to provide a lightweight PaaS with a Heroku-inspired workflow.
+
+## 1.4.0 Summary
+
+### New Features
+
+- You can now provision Deis on [Azure](http://azure.microsoft.com/)! - [#2893](https://github.com/deis/deis/pull/2893)
+- AWS provisioning scripts support Elastic Load Balancers - [#3074](https://github.com/deis/deis/pull/3074)
```
Comment (carmstrong) [inline _posts/2015-03-03-deis-1-4-now-on-azure.md]: We've always supported ELBs - they now support *internal* ELBs.

File: _posts/2015-03-03-deis-1-4-now-on-azure.md [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+- **router** allows optional HTTPS redirect - [#3075](https://github.com/deis/deis/pull/3075)
+- **store-gateway** can be scaled with `deisctl scale` - [#2929](https://github.com/deis/deis/pull/2929)
+- Any scalable component can be started or stopped with `deisctl start|stop component@` - [#3106](https://github.com/deis/deis/pull/3106)
+- New documentation added about [disk usage](http://docs.deis.io/en/latest/managing_deis/disk_usage/) - [#3021](https://github.com/deis/deis/pull/3021)
+- New links added to Deis [community projects](https://github.com/deis/deis/blob/master/contrib/README.md#community-contributions)! - [#3114](https://github.com/deis/deis/pull/3114)
+
+### Improvements
+
+- **builder** properly escapes backticks in environment variables - [#3129](https://github.com/deis/deis/pull/3129)
+- **builder** lets docker-in-docker choose its storage driver - [#3039](https://github.com/deis/deis/pull/3039)
```
Comment (carmstrong) [inline _posts/2015-03-03-deis-1-4-now-on-azure.md]: Nitpick, but shouldn't Docker-in-Docker be capitalized?

**Comment by carmstrong** [comment]
LGTM after comments.

**Pull request merged**